integration-tests/bats/auto_increment_lock_modes.bats

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env bats
+load $BATS_TEST_DIRNAME/helper/common.bash
```

Comment:
Did you get a chance to check out the sql server multi session tests in Dolt's Go code? Those tests are somewhat easier to debug, although since they start up a separate sql server process, it's still not as direct as a non-server test. They also have the advantage of having multiple concurrent sessions running and being able to interleave commands from different sessions. They are a little hacky, but improving them so we could use them more would be pretty valuable.